features/upgrade/samples_operator/upgrade.feature

```diff
@@ -37,3 +37,99 @@ Feature: image-registry operator upgrade tests
     Given the expression should be true> cluster_operator('openshift-controller-manager').condition(type: 'Available')['status'] == "True"
     Given the expression should be true> cluster_operator('openshift-controller-manager').condition(type: 'Progressing')['status'] == "False"
     Given the expression should be true> cluster_operator('openshift-controller-manager').condition(type: 'Degraded')['status'] == "False"
+
+  @upgrade-prepare
+  @users=upuser1,upuser2
+  @admin
+  Scenario: Upgrade imagestream from old version which not implement node credentials - prepare 
+    Given I switch to cluster admin pseudo user
+    When I run the :extract admin command with:
+      | resource  | secret/pull-secret |
+      | namespace | openshift-config   | 
+      | to        | /tmp               |
+      | confirm   | true               |
+    Then the step should succeed
+    When I run the :new_project client command with:
+      | project_name | devexp-p1 |
+    Then the step should succeed
+    When I use the "devexp-p1" project
+    When I run the :create_secret client command with:
+      | secret_type | generic                                  |
+      | name        | pj-secret                                |
+      | from_file   | .dockerconfigjson=/tmp/.dockerconfigjson |
+      | type        | kubernetes.io/dockerconfigjson           |
+    Then the step should succeed
+    When I run the :import_image client command with:
+      | from             | registry.redhat.io/rhscl/ruby-25-rhel7:latest |
+      | confirm          | true                                          |
+      | image_name       | imagestream1:latest                           |
+      | reference-policy | local                                         |
+    Then the step should succeed
+    And the output should not contain:
+      | error: Import failed |
+    When I run the :import_image client command with:
+      | from             | registry.redhat.io/rhscl/ruby-25-rhel7:latest |
+      | confirm          | true                                          |
+      | image_name       | imagestream2:latest                           |
+    Then the step should succeed
+    And the output should not contain:
+      | error: Import failed |
```

Comment:
Do we need string checks like this? They are unreliable check IMO and only make scenario longer.

I'd suggest to remove this and all other similar string checks unless you have seen actual value from them.

Reply:
other than that lgtm

Reply:
@akostadinov Import_image cmd return succeed whenever the image import succeed or fail. In result we need check if there are import failure to confirm the image import situation.

Reply:
@xiuwang , this deserves a bugzilla. A failed command should return a non-zero status. I'll merge but this is a bug.